code/datums/components/shell.dm

```diff
@@ -181,3 +187,19 @@
 		attached_circuit.remove_component(to_remove)
 		to_remove.moveToNullspace()
 	attached_circuit = null
+
+/datum/component/shell/proc/on_atom_usb_cable_try_attach(datum/source, obj/item/usb_cable/usb_cable, mob/user)
+	SIGNAL_HANDLER
+
+	var/atom/atom_parent = parent
```

Comment:
Might want to just change datum/source to be atom/atom_parent and get rid of this variable. Minor optimization, would be good for code quality though to use the source.